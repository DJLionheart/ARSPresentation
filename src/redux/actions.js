import fetch from '../utils/fetch'

export const setLuke = (data) => ({
    type: types.LUKE,
    payload: data
})

export const fetchLuke = () =>
    async (disaptch) => {
        const response = await disaptch(fetch.get('https://swapi.co/api/people/1'))
        console.log('hit', response)
        disaptch(setLuke(response))
    }

export const types = {
    LUKE: 'LUKE'
}