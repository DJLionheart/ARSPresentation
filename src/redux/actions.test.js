import nock from 'nock'
import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import { Map } from 'immutable'

import * as actions from './actions'

describe('actions', () => {
    const mockStore = configureStore([thunk])
    let store
    const state = Map()
    beforeEach(() => {
        store = mockStore(state)
    })

    afterEach(() => {
        store.clearActions()
        nock.cleanAll()
    })

    test('action.setLuke', () => {
        const payload = {
            name: 'Luke Skywaler',
            from: 'Swapi'
        }

        expect(actions.setLuke(payload)).toMatchSnapshot()
    })

    test('actions.fetchLuke', async () => {
        const response = {
            name: 'Luke Skywaler',
            from: 'Swapi'
        }

        nock('https://swapi.co/api')
            .get('/people/1')
            .reply(200, response)

        await store.dispatch(actions.fetchLuke())
        expect(store.getActions()).toMatchSnapshot()
    })
})