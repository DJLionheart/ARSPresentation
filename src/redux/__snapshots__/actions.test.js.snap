// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions action.setLuke 1`] = `
Object {
  "payload": Object {
    "from": "Swapi",
    "name": "Luke Skywaler",
  },
  "type": "LUKE",
}
`;

exports[`actions actions.fetchLuke 1`] = `
Array [
  Object {
    "payload": Object {
      "from": "Swapi",
      "name": "Luke Skywaler",
    },
    "type": "LUKE",
  },
]
`;
